slither/solc_parsing/declarations/contract.py

```diff
@@ -224,7 +224,9 @@ def parse_structs(self):
     def parse_state_variables(self):
         for father in self.inheritance_reverse:
             self._variables.update(father.variables_as_dict())
-            self._variables_ordered += father.state_variables_ordered
+            for var in father.state_variables_ordered:
```

Comment:
Do we have an example where this could lead to a duplicate?

If so it might be worth looking why it happens, it might hide some weird bugs in another location

Reply:
For the following contract:
```
contract A1 {
    uint a1;
}

contract B1 is A1 {
    uint b1;
}

contract B2 is A1 {
    uint b2;
}

contract C1 is B1, B2 {
    uint c1;
}

contract D1 is C1 {
    uint d1;
}
```

On master:
```
D1:
+-------+---------+
|  Name |   Type  |
+-------+---------+
| A1.a1 | uint256 |
| A1.a1 | uint256 |
| B1.b1 | uint256 |
| A1.a1 | uint256 |
| B2.b2 | uint256 |
| A1.a1 | uint256 |
| A1.a1 | uint256 |
| B1.b1 | uint256 |
| A1.a1 | uint256 |
| B2.b2 | uint256 |
| C1.c1 | uint256 |
| D1.d1 | uint256 |
+-------+---------+
```

On this branch
```
D1:
+-------+---------+------+--------+
|  Name |   Type  | Slot | Offset |
+-------+---------+------+--------+
| A1.a1 | uint256 |  0   |   0    |
| B1.b1 | uint256 |  1   |   0    |
| B2.b2 | uint256 |  2   |   0    |
| C1.c1 | uint256 |  3   |   0    |
| D1.d1 | uint256 |  4   |   0    |
+-------+---------+------+--------+
```